bayesian/utility/v2/sa_response_builder.py

```diff
@@ -0,0 +1,106 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Stack analyses API v2 response builder class."""
+
+import logging
+from bayesian.exceptions import HTTPError
+from bayesian.utils import request_timed_out
+
+logger = logging.getLogger(__file__)
+
+
+class StackAnalysesResponseBuilder:
+    """Stack analysis response builder for API v2.
+
+    Class takes db result, stack result and recommendation data to verify and build stack
+    analyses response for V1.
+    """
+
+    def __init__(self, external_request_id, rdb_analyses):
+        """Response Builder, Build Json Response for Stack Analyses."""
+        self.external_request_id = external_request_id
+        self.rdb_analyses = rdb_analyses
+
+    def get_response(self):
+        """Aggregate, build and return json response for the given request id."""
+        # Get db result, stack result and recm data from rdb.
+        self._db_result = self.rdb_analyses.get_request_data()
+        self._stack_result = self.rdb_analyses.get_stack_result()
+        self._recm_data = self.rdb_analyses.get_recommendation_data()
+
+        # If request is invalid, it will raise HTTPError with proper message.
+        self._raise_if_invalid()
+
+        # If request is inprogress or timeout, it will raise HTTPError with proper message.
+        self._is_request_inprogress()
+
+        # Proceed with building actual response from data.
+        stack_task_result = None
+        stack_audit = None
+        recommendation = {}
+
+        if self._stack_result is not None and 'task_result' in self._stack_result:
```

Comment:
Do we have a situation where self._stack_result is not None but it doesn't have `task_result` in it?

Reply:
Don't know, it is object return from backbone server, so added this check for extra safety. Do you see any adverse effect of this extra checks?

Reply:
In that case assertion would be the right choice. Having many conditions hinders readability and testability.

Reply:
I will revert to only None check for stack and recommendation response.